unit-tests/utilities/concurrency/test-dispatcher.cpp

```diff
@@ -0,0 +1,91 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+//#cmake:add-file ../../../src/dispatcher.cpp
+
+#include <unit-tests/test.h>
+#include <common/utilities/time/timer.h>
+#include <src/concurrency.h>
+
+#include <algorithm>
+#include <vector>
+
+using namespace utilities::time;
+
+// We use this function as a CPU stress test function
+int fibo( int num )
+{
+    if( num < 2 )
+        return 1;
+    return fibo( num - 1 ) + fibo( num - 2 );
+}
+
+TEST_CASE( "dispatcher main flow" )
+{
+    dispatcher d(2);
+    std::atomic_bool run = { false };
+    auto func = [&](dispatcher::cancellable_timer c) 
+    {
+        run = true;
+    };
+
+    d.start();
+    REQUIRE(d.empty());
+    d.invoke(func);
+    REQUIRE_FALSE(d.empty());
+    REQUIRE(d.flush());
+    REQUIRE(run);
+    d.stop();   
+}
+
+TEST_CASE( "invoke and wait" )
+{
+    dispatcher d(2);
+
+    std::atomic_bool run = { false };
+    auto func = [&](dispatcher::cancellable_timer c)
+    {
+        std::this_thread::sleep_for(std::chrono::seconds(3));
```

Comment:
This should really be using the cancellable_timer... and then we can test that, too... something to think about :)